tests/functional/conftest.py

```diff
@@ -212,21 +211,32 @@ def _check(
 @pytest.fixture
 def wait_for_sidekiq(gl):
     """
-    Return a helper function to wait until there are no busy sidekiq processes.
+    Return a helper function to wait until there are spare sidekiq processes.
+
+    Because not all Groups can be deleted immediately in GL 16, we will likely have busy Sidekiq jobs
+    set aside for their deletion in 1 days time.
 
     Use this with asserts for slow tasks (group/project/user creation/deletion).
+
+    { ..., 'labels': [], 'concurrency': 20, 'busy': 3}
```

Comment:
@TimKnight-DWP I think I introduced this sidekiq helper years ago thinking it's a bit smarter than simple sleep calls. But I wonder if we should just delete this and instead have simpler helpers that wait for a resource to be in a certain state. E.g. wait for branch to exist/not exist, wait for MR to have certain status etc.

I'm just not sure how much work it would take, but it might be simpler.

Reply:
I'll have a look, yeah the ones it mainly seems to matter for now are Merging MRs and I'm having to put more specific logic

I have John's MR open in a tab to guide me when I start on the `api` tests 👍